### PR TITLE
Fix some doc URLs in catalog

### PR DESCRIPTION
**Pull request by sir-sigurd**
It looks like URLs were changed by c5ededee0a54c978f47b8eba137e41584ba6d993. There are redirects to new locations but URL fragment is not preserved after redirect.

This change is mostly to fix URLs with fragments, others are updated for uniformity.

### CODE REVIEW

File: catalog/app/containers/Bucket/errors.tsx [hunk 1]
```diff
@@ -167,7 +167,7 @@ const defaultHandlers: ErrorHandler[] = [
           <br />
           <StyledLink
             target="_blank"
-            href={`${docs}/walkthrough/working-with-the-catalog#access-control`}
+            href={`${docs}/quilt-platform-administrator/admin#users-and-roles`}
```
Comment (sir-sigurd) [inline catalog/app/containers/Bucket/errors.tsx:170]: this anchor was broken in https://github.com/quiltdata/quilt/commit/2c1aec8a5f5db4665db54fabeae5d61f9630f309#diff-c83d883e5d164b4035b039bb964282da0e39d8bafa72e2b193cd6760d8b976b1

I'm not sure if it should refer to the old place: https://docs.quiltdata.com/quilt-platform-catalog-user/working-with-the-catalog#admin-ui

**Comment by drernie** [comment]
Ouch, sorry about that. Fixes approved.

**Review by drernie — APPROVED** [review]

**Review by nl0 — DISMISSED** [review]
lgtm. a changelog would be nice

File: catalog/app/containers/Bucket/errors.tsx (hunk 1 above)
Comment (nl0) [reply catalog/app/containers/Bucket/errors.tsx:170]: i think it's ok like this

**Comment by drernie** [comment]
@sir-sigurd Are you just needing to add the CHANGELOG?

**Review by nl0 — APPROVED** [review]

**Pull request merged**